jvm/ml4ir-inference/src/test/integration/ml4ir/inference/tensorflow/TensorFlowInferenceIT.scala

```diff
@@ -53,11 +89,109 @@ class TensorFlowInferenceIT extends TestData {
     return Source.fromFile(csvFile).getLines.toList.tail.flatMap(extractColumnValues)
   }
 
+  def validateRankingScores(query: StringMapQueryAndPredictions,
+                     sequenceExample: SequenceExample,
+                     scores: Array[Float],
+                     numDocs: Int) = {
+    val docScores = scores.take(numDocs)
+    val maskedScores = scores.drop(numDocs)
+    docScores.foreach(
+      score => assertTrue("all docs should score non-negative", score > 0)
+    )
+    for {
+      maskedScore <- maskedScores
+      docScore <- docScores
+    } {
+      assertTrue(
+        s"docScore ($docScore) should be > masked score ($maskedScore)",
+        docScore > maskedScore
+      )
+    }
+    if (query.predictedScores != null) {
+      assertArrayEquals("scores aren't close enough: ", docScores, query.predictedScores, 1e-2f)
+    }
+
+  }
+
+  def validateScores(scores: Array[Float], numDocs: Int) = {
+    val docScores = scores.take(numDocs)
+    val maskedScores = scores.drop(numDocs)
+    docScores.foreach(
+      score => assertTrue("all docs should score non-negative", score > 0)
+    )
+    for {
+      maskedScore <- maskedScores
+      docScore <- docScores
+    } {
+      assertTrue(
+        s"docScore ($docScore) should be > masked score ($maskedScore)",
+        docScore > maskedScore
+      )
+    }
+    assertTrue(
+      "second doc should score better than first",
+      scores(1) > scores(0)
+    )
+    println(scores.mkString(", "))
+  }
+
+  @Test
+  def testRankingSavedModelBundleWithCSVData(): Unit = {
+    val generatedBundleLocation = System.getProperty("bundleLocation")
+    def modelName = System.getProperty("runName") + "_ranking"
```

Comment:
we should have a non-@Test annotated method which takes in bundleLocation and runName as method parameters and does what this method does after these lines.  Then the @Test method just grabs the System properties and executes this method with them.

That way, we can easily change it to run not using system properties, but instead commandline args, or having the training script write pairs of paths to a file, or whatever.

Reply:
I added this. But I didn't add (yet) a bootstrap class to have this run stand-alone.

Reply:
@ducouloa the bootstrap still getting added? If so, please add a quick paragraph about it at whichever would be the appropriated place at the docs.

Reply:
@FranciscoBorges There is W-9280938 to follow-up.